apps/internal/base/base.go

```diff
@@ -123,11 +123,13 @@ func New(clientID string, authorityURI string, cacheAccessor cache.ExportReplace
 		return Client{}, err
 	}
 	authParams := authority.NewAuthParams(clientID, authInfo)
-
+	if cacheAccessor == nil {
```

Comment:
might be worth documenting this in the comment for the method

Reply:
The other option is to make this a Go style option:

```go
func Option func(c *Client)

type WithCacheAccessor(ca cache.ExportReplace) Option {
  return func(c *Client) {
    if c == nil {
      return
    }
    c.cacheAccessor = ca
}

func New(clientID, authorityURI, token *oauth.Client, options ...Option) (Client, error) {
  ...
  client := Client {
    ...
   cacheAccessor: noopCacheAccessor{}
  }
  for _, o := range options {
    o(&client)
  }
  return client, nil
```

Then its a true option (which I prefer over accepting nils).

Reply:
@element-of-surprise , Tried changing this to accept it as an option in the latest commit.  
What do you think ?

Reply:
Looks better, thanks

Reply:
Merging this in.